rust/rsc/src/bin/rsc/read_job.rs

```diff
@@ -63,152 +63,130 @@ pub async fn read_job(
+    let stdout_blob = match resolve_blob(matching_job.stdout_blob_id, conn.as_ref(), &blob_stores).await {
+            Err(err) => {
+                tracing::error! {%err, "Failed to resolve stdout blob. Resolving job as a cache miss."};
+                return (StatusCode::NOT_FOUND, Json(ReadJobResponse::NoMatch));
+            },
+            Ok(blob) => blob,
+    };
 
-                let stdout_blob = match resolve_blob(matching_job.stdout_blob_id, txn, &blob_stores).await {
-                    Err(err) => {
-                        tracing::error! {%err, "Failed to resolve stdout blob. Resolving job as a cache miss."};
-                        return Ok((None, ReadJobResponse::NoMatch))
-                    },
-                    Ok(blob) => blob,
-                };
+    let stderr_blob = match resolve_blob(matching_job.stderr_blob_id, conn.as_ref(), &blob_stores).await {
+            Err(err) => {
+                tracing::error!(%err, "Failed to resolve stderr blob. Resolving job as a cache miss.");
+                return (StatusCode::NOT_FOUND, Json(ReadJobResponse::NoMatch));
+            },
+            Ok(resolved_blob) => resolved_blob,
+    };
 
-                let stderr_blob = match resolve_blob(matching_job.stderr_blob_id, txn, &blob_stores).await {
-                    Err(err) => {
-                        tracing::error! {%err, "Failed to resolve stderr blob. Resolving job as a cache miss."};
-                        return Ok((None, ReadJobResponse::NoMatch))
-                    },
-                    Ok(blob) => blob,
-                };
+    // Step 5: Construct response
+    let response = ReadJobResponse::Match {
+        output_symlinks,
+        output_dirs,
+        output_files,
+        stdout_blob,
+        stderr_blob,
+        status: matching_job.status,
+        runtime: matching_job.runtime,
+        cputime: matching_job.cputime,
+        memory: matching_job.memory as u64,
+        ibytes: matching_job.i_bytes as u64,
+        obytes: matching_job.o_bytes as u64,
+    };
 
-                Ok((
-                    Some(matching_job.id),
-                    ReadJobResponse::Match {
-                        output_symlinks,
-                        output_dirs,
-                        output_files,
-                        stdout_blob,
-                        stderr_blob,
-                        status: matching_job.status,
-                        runtime: matching_job.runtime,
-                        cputime: matching_job.cputime,
-                        memory: matching_job.memory as u64,
-                        ibytes: matching_job.i_bytes as u64,
-                        obytes: matching_job.o_bytes as u64,
-                    },
-                ))
-            })
-        })
-        .await;
+    // Step 6: TODO before recording a hit and returning, need verification that objects from transaction did not change in
+    // database, or convert a hit into a miss
```

Comment:
I have questions on how best to accomplish this verification from chaning a hit to a miss. Do I just query (output_files, output_symlinks, output_dirs) again and verify they are the same with the database? Seems kind of wasteful (what happens when another update happens while I am doing the verification equivalence check?). 

Can we instead potentially rely on job::Entity's created_at field (or create a new one called updated_at) where we just quickly check timestamps are same before returning?